packages/api/src/users/users.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@nestjs/common';
+
+import { RegisteredUser, User } from '@coderscamp/shared/models';
+
+import { UsersRepository } from './users.repository';
+import { UserFromJwt } from './users.types';
+
+@Injectable()
+export class UsersService {
+  constructor(private readonly usersRepository: UsersRepository) {}
+
+  async update(user: Partial<User> & UserFromJwt): Promise<User | RegisteredUser> {
+    const userFromDb = await this.usersRepository.getById(user.id);
+
+    if (!userFromDb) {
+      throw new Error('User not found');
```

Comment:
💭 `throw new NotFoundError()` would be better than creating custom errors

Reply:
NotFound error is strictly HTTP exception, here we are on lower layer (business logic). NotFound is indeed better, but maybe in controller? Imagine that you're interact with this business logic by messaging or gRPC, then NotFound - HTTP 404 is not accurrate.

Reply:
The question is, wouldn't it be easier to care about other communication protocols when we actually plan to implement them?

Reply:
It depends... - as always :D 
But for me, it's similar like exposing database entity from rest api = mixing layers.